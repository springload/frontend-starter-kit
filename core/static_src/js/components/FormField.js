import React from 'react';

// eslint-disable-next-line react/prefer-es6-class
const FormField = React.createClass({
    propTypes: {
        id: React.PropTypes.string.isRequired,
        name: React.PropTypes.string,
        value: React.PropTypes.oneOfType([
            React.PropTypes.string,
            React.PropTypes.number,
        ]).isRequired,
        label: React.PropTypes.string.isRequired,
        type: React.PropTypes.string,
        error: React.PropTypes.string,
        onChange: React.PropTypes.func.isRequired,
        onEnterKey: React.PropTypes.func,
        autoCorrect: React.PropTypes.string,
        autoCapitalize: React.PropTypes.string,
        spellCheck: React.PropTypes.string,
        errorClass: React.PropTypes.string,
        defaultValue: React.PropTypes.string,
    },

    getDefaultProps() {
        return {
            onEnter: null,
            name: null,
            type: 'text',
            error: null,
            onEnterKey: null,
            autoCorrect: 'off',
            autoCapitalize: 'off',
            spellCheck: 'off',
            errorClass: 'theme__text-invert',
            defaultValue: null,
        };
    },

    handleChange(e) {
        if (this.props.onChange) {
            this.props.onChange(this.props.id, e.target.value);
        }
    },

    handleKeyDown(e) {
        if (this.props.onEnterKey) {
            if (e.keyCode === 13) {
                this.props.onEnterKey(e);
            }
        }
    },

    render() {
        const className = ['theme__color', 'theme__border'];
        if (this.props.error) {
            className.push('error');
        }

        const classError = `form-field__error ${this.props.errorClass}`;

        return (
            <div className="form-field">
                <label
                    className="h6"
                    htmlFor={this.props.id}
                >
                    {this.props.label}
                </label>
                <input
                    className={className.join(' ')}
                    ref={this.props.id}
                    id={this.props.id}
                    name={this.props.name}
                    type={this.props.type}
                    value={this.props.value}
                    onChange={this.handleChange}
                    onBlur={this.handleBlur}
                    onFocus={this.handleFocus}
                    onKeyDown={this.handleKeyDown}
                    autoCorrect={this.props.autoCorrect}
                    autoCapitalize={this.props.autoCapitalize}
                    spellCheck={this.props.spellCheck}
                    defaultValue={this.props.defaultValue}
                />
                {this.props.error ? (<div className={classError}>{this.props.error}</div>) : null}
            </div>
        );
    },

});

export default FormField;
