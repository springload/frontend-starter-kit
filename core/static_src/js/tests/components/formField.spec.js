import renderer from 'react-test-renderer';
import FormField from '../../components/FormField';
import React from 'react';

describe('FormField', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <FormField id="field1" name="field1" value="Hello world!" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
