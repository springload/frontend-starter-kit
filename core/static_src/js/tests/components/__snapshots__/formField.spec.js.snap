exports[`FormField renders correctly 1`] = `
<div
  className="form-field">
  <label
    className="h6"
    htmlFor="field1" />
  <input
    autoCapitalize="off"
    autoCorrect="off"
    className="theme__color theme__border"
    defaultValue={undefined}
    id="field1"
    name="field1"
    onBlur={undefined}
    onChange={[Function]}
    onFocus={undefined}
    onKeyDown={[Function]}
    spellCheck="off"
    type="text"
    value="Hello world!" />
</div>
`;
